geomstats/geometry/base.py

```diff
@@ -12,6 +12,24 @@
 from geomstats.geometry.pullback_metric import PullbackMetric
 
 
+class MathStruct(abc.ABC):
```

Comment:
Let's just class it Structure, to avoid concatenating abbreviations which end up being hard to read for external users & collaborators.

Reply:
`MathematicalStructure` instead or would be too long?